adr/001-component-library.md

```diff
@@ -0,0 +1,60 @@
+# Introduce component library
+
+- Status: Proposed
+- Deciders: Team
+- Date: 17.10.2024
+
+## Result
+
+A1: Component library is introduced, firstly as just a folder in our current setup, adding yarn workspaces or similar requires more research and testing
+
+## Problem context
+
+Today our UI components are tightly coupled to the app in which they are rendered.
+
+This leads to several issues:
+
+- Makes it hard to do component testing outside a fully rendered app.
+- Makes refactoring the app framework a lot harder.
+- Leads to unclear interfaces between UI components and the app framework.
```

Comment:
Would be nice to have linked examples of code/components for some of these points, I'm sure you've discussed and looked at tons of examples here to inform this assertion but for myself presently and future readers it would be helpful to know the context here 😄

Reply:
Agree, done!